src/main/java/pw/itr0/kaba/encrypt/PasswordBasedDecrypter.java

```diff
@@ -12,27 +17,61 @@
  * @since 1.0.0
  */
 public interface PasswordBasedDecrypter {
+
     /**
-     * Decrypt {@code secret} based on password.
+     * Returns {@link Cipher} to decrypt bytes with {@code password}.
      *
      * @param password decrypting password
+     * @return {@link Cipher} to be used on decrypt
+     * @throws InvalidKeyException if the {@code password} does not satisfy password specification of using decryption algorithm,
+     *                             or decryption algorithm using specified password length is not usable on current JDK
+     */
+    Cipher getDecryptionCipher(char[] password) throws InvalidKeyException;
+
+    /**
+     * Decrypt {@code secret} based on password.
+     *
      * @param secret   secret bytes
+     * @param password decrypting password
      * @return decrypted secret
-     * @throws GeneralSecurityException on decryption failure
+     * @throws InvalidKeyException       if the {@code password} does not satisfy password specification of using decrypting algorithm,
+     *                                   or decrypting algorithm using specified password length is not usable on current JDK
+     * @throws BadPaddingException       if the {@code secret} is not properly padded
+     * @throws IllegalBlockSizeException if the {@code secret} has wrong length
      */
-    byte[] decrypt(char[] password, byte[] secret) throws GeneralSecurityException;
+    default byte[] decrypt(byte[] secret, char[] password)
+            throws InvalidKeyException, BadPaddingException, IllegalBlockSizeException {
+        return getDecryptionCipher(password).doFinal(secret);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method call to 'getDecryptionCipher' needs "this.".](https://www.codacy.com/app/ryotan/kaba-vault/file/2686265516/issues/source?bid=3298589&fileBranchId=3332567#l44)